controllers/autohealconfig_controller.go

```diff
@@ -47,16 +66,415 @@ type AutohealConfigReconciler struct {
+			if configmap.Data == nil {
+				configmap.Data = make(map[string]string)
+			}
+			configmap.Labels = map[string]string{
+				"app": "autoheal",
+			}
+			configmap.Data["query"] = query
+			configmap.Data["label"] = label
+			configmap.Data["script"] = script
+			configmap.Data["gracePeriod"] = graceperiod
+			return nil
+
+		})
+		if err != nil {
+			log.Errorf("Unable to create config map")
+			return err
+
+		}
+	}
+	return nil
+}
+
+// Updates the controller to how long until next requeue of the reconciller depening on if its a error or not
+func (r *AutohealConfigReconciler) updateStatus(cr *v1alpha1.AutohealConfig) (ctrl.Result, error) {
+	err := r.Client.Status().Update(context.Background(), cr)
+	if err != nil {
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: RequeueDelayError,
+		}, err
+	}
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: RequeueDelaySuccess,
+	}, nil
+}
+
+// Delests the config secret with creds to sync to the repo
+func (r *AutohealConfigReconciler) deleteConfigSecret(config_cr *v1alpha1.AutohealConfig, ctx context.Context) error {
+	configRepoSecret := &core.Secret{}
+	secretSelector := &client.ObjectKey{
+		Namespace: config_cr.Namespace,
+		Name:      AutohealConfigSecret,
+	}
+	err := r.Client.Get(ctx, *secretSelector, configRepoSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Cannot find config map")
+
+		}
+		log.Errorf("Error cant get config repo secret")
+		return err
+
+	}
+	log.Info(" Autoheal-operator, Stage Clean up: Deleting config repo secret")
+	err = r.Client.Delete(ctx, configRepoSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Cannot find config secret")
+			return nil
+
+		}
+		log.Errorf("Error getting secret for deletion")
+		return err
+	}
+
+	return nil
+}
+
+// Delets all configs for cleanup
+func (r *AutohealConfigReconciler) deleteConfigmap(config_cr *v1alpha1.AutohealConfig, ctx context.Context) error {
+	configmap := &core.ConfigMapList{}
+	configSelector := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"app": "autoheal",
+		}),
+		Namespace: config_cr.Namespace,
+	}
+
+	err := r.Client.List(ctx, configmap, configSelector)
+	if err != nil {
+		log.Errorf("Error cant get config repo secret")
+		return err
+	}
+	log.Info(" Autoheal-operator, Stage Clean up: Deleting config maps")
+	for _, configs := range configmap.Items {
+		err = r.Client.Delete(ctx, &configs)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Info("Cannot find config map: ", configs)
+
+			}
+			log.Errorf("Error cannnot delete configmap")
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Deletes unwanted config maps i.e if the config repo has been changed to remove a scenerio it is removed from the cluster
+func (r *AutohealConfigReconciler) removeUnwantedCM(ctx context.Context, scenarios *[]v1alpha1.AutohealConfigScenario) error {
```

Comment:
```suggestion
func (r *AutohealConfigReconciler) removeUnwantedConfigMap(ctx context.Context, scenarios *[]v1alpha1.AutohealConfigScenario) error {
```
Just to keep the function naming conventions in line with each other, as we use `ConfigMap` explicitly in other function names.